linter/check_cmake_layout.py

```diff
@@ -0,0 +1,30 @@
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from astroid import nodes
+
+
+class CmakeLayout(BaseChecker):
+    """
+    Do not add src_folder to cmake_layout
+    """
+
+    __implements__ = IAstroidChecker
+
+    name = "conan-cmake-layout"
+    msgs = {
+        "W9012": (
```

Comment:
Since the majority of upgraded recipes are already doing this I think it's good to very strongly push it to be in all new ones as well

```suggestion
        "E9012": (
```